android/app/src/main/java/com/cBreez/client/BreezForegroundService.kt

```diff
@@ -0,0 +1,196 @@
+    }
+
+    // =========================================================== //
+    //                      SERVICE LIFECYCLE                      //
+    // =========================================================== //
+
+    override fun onBind(intent: Intent?): IBinder {
+        Logger.tag(TAG).debug { "Binding Breez node service from intent=$intent" }
+        paymentHashPollerList.clear()
+        stopForeground(STOP_FOREGROUND_REMOVE)
+        dismissForegroundServiceNotification(applicationContext)
+        return binder
+    }
+
+    override fun onUnbind(intent: Intent?): Boolean {
+        return false
+    }
+
+    private val shutdownHandler = Handler(Looper.getMainLooper())
+    private val shutdownRunnable: Runnable = Runnable {
+        Logger.tag(TAG).debug { "Reached scheduled shutdown..." }
+        if (paymentHashPollerList.isEmpty()) {
+            stopForeground(STOP_FOREGROUND_REMOVE)
+        } else {
+            stopForeground(STOP_FOREGROUND_DETACH)
+        }
+        shutdown()
+    }
+
+    /** Shutdown the node, close connections and stop the service */
+    private fun shutdown() {
+        Logger.tag(TAG).info { "Shutting down Breez node service" }
+        stopSelf()
+    }
+
+    // =========================================================== //
+    //                    START COMMAND HANDLER                    //
+    // =========================================================== //
+
+    /** Called when an intent is called for this service. */
+    @RequiresApi(Build.VERSION_CODES.TIRAMISU)
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        super.onStartCommand(intent, flags, startId)
+        Logger.tag(TAG).debug {
+            "Start Breez node service from intent [ intent=$intent, flag=$flags, startId=$startId ]"
+        }
+        intent.getRemoteMessage()?.let {
+            if (it.data["notification_type"] == "payment_received") {
+                val paymentHash = it.data["payment_hash"]
+                val clickAction = it.data["click_action"]
+                paymentHash?.let {
+                    val notification = notifyForegroundService(applicationContext)
+                    if (breezSDK == null) {
+                        startForeground(NOTIFICATION_ID_FOREGROUND_SERVICE, notification)
+                        breezSDK = connectSDK(applicationContext, SDKListener())
+                    }
+                    pollerScope.launch(
+                        Dispatchers.IO +
+                                CoroutineExceptionHandler { _, e ->
+                                    Logger.tag(TAG).error { "Error when polling payment: $e" }
+                                    notifyPaymentFailed(applicationContext)
+                                }
+                    ) {
+                        pollPaymentHash(paymentHash, clickAction)
+                    }
+                }
+            }
+        }
+        return START_NOT_STICKY
```

Comment:
Seems like we should schedule a shutdown here otherwise there are cases where the service will not stop.

Reply:
Addressed as part of [Remove payment hash poller & everything related to failed payment not…](https://github.com/breez/c-breez/pull/748/commits/fc1bb41f68315a4d63fbac3955c802d002d00a73)